### PR TITLE
Subgraph Refactor

### LINKED ISSUE #618: [SUBGRAPH] Refactor Subgraph Tests
- Refactor functions to be more general
- Try to simplify code more
- Remove unnecessary contracts (leverage sdk-core more)

### PR DESCRIPTION
**Pull request by 0xdavinchee**

### CODE REVIEW

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -1028,9 +1030,9 @@ interaction with a `token`.
 """
 type AccountTokenSnapshot @entity {
     """
-    ID composed of: accountID-tokenID
+    ID composed of: accountID + keccak256("-") + tokenID (without the 0x)
     """
-    id: ID!
+    id: Bytes!
```
Comment (kasparkallas) [inline packages/subgraph/schema.graphql]: This does not look right... ? The field is `Bytes` but it's actually a concatenated string.

File: packages/subgraph/schema.graphql [hunk 2]
```diff
@@ -1094,9 +1096,9 @@ TokenStatistic: An aggregate entity which aggregates data of a single `token`.
 """
 type TokenStatistic @entity {
     """
-    ID: tokenID
+    id: token address
     """
-    id: ID!
+    id: Bytes!
```
Comment (kasparkallas) [inline packages/subgraph/schema.graphql]: Same here. "id" should always be "ID", no?

File: packages/subgraph/src/mappings/idav1.ts [hunk 3]
```diff
@@ -423,7 +418,7 @@ export function handleSubscriptionUnitsUpdated(
     if (!hasValidHost) {
         return;
     }
-    let tokenId = event.params.token.toHex();
+    let tokenId = event.params.token;
```
Comment (kasparkallas) [inline packages/subgraph/src/mappings/idav1.ts]: Does not look like this is used?

**Review by kasparkallas — CHANGES_REQUESTED** [review]

**Comment by hellwolf** [comment]
@msoni89 please also have a look

File: packages/subgraph/src/mappingHelpers.ts [hunk 4]
```diff
@@ -82,7 +78,7 @@ export function getOrInitSuperToken(
     let currentTimestamp = block.timestamp;
     let resolverAddress = getResolverAddress();
 
-    if (tokenId == ZERO_ADDRESS.toHex()) {
+    if (tokenAddress.equals(ZERO_ADDRESS)) {
```
Comment (msoni89) [inline packages/subgraph/src/mappingHelpers.ts:78]: Please early return it, is possible

Comment (0xdavinchee) [reply packages/subgraph/src/mappingHelpers.ts:78]: done

**Comment by hellwolf** [comment]
@kasparkallas @msoni89 please revisit

**Review by kasparkallas — APPROVED** [review]

**Pull request merged**